samples/analyze/analyze_test.py

```diff
@@ -54,13 +58,19 @@ def test_analyze_labels_file(capsys):
     out, _ = capsys.readouterr()
     assert "label description" in out
 
-
 @pytest.mark.slow
 def test_analyze_explicit_content(capsys):
-    analyze.analyze_explicit_content("gs://cloud-samples-data/video/cat.mp4")
-    out, _ = capsys.readouterr()
-    assert "pornography" in out
-
+    while True:
+        try:
+            analyze.analyze_explicit_content("gs://cloud-samples-data/video/cat.mp4")
+            out, _ = capsys.readouterr()
+            assert "pornography" in out
+        except ServiceUnavailable as e:
+            # Service is throttling or not available for the moment, sleep for 5 sec and retrying again.
+            print("Got service unavailable exception: {}".format(str(e)))
+            time.sleep(5)
+            continue
+        break
```

Comment:
Is it possible that the test can run almost indefinitely if it continues to encounter `ServiceUnavailable`? If not and it fails after few minutes I think that's fine, but otherwise would it be possible to have some sort of reasonable ceiling for it if applicable rather than an infinite loop?

Reply:
that sounds reasonable,  I just converted into infinite loop into for loop.